### PR TITLE
Patch 1.2.2

### PR DESCRIPTION
**Pull request by itzCozi**
Hello its me again, good progress man I've fixed the usual spelling, formatting, bad-practices and indentation along with others like reworking how you exit the program and call functions only when their needed. A `colors` class has been added to contain all the color variables because they are all similar. The `README.md` was updated and changes are now displayed there along with a detailed and organized installation guide

### CODE REVIEW

File: source/converter.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+'''
+
+# This is a class containing variables of similar values
+class colors:
+  # To use a variable in this class calls the class name then the variable  
+  blue = os.system("color b")
+  red = os.system("color C")
+  green = os.system("color a")
+  pink = os.system("color D")
```
Comment (itzCozi) [inline source/converter.py:38]: Put functions in class to organize

File: source/converter.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+class functions:
+
+  # Create a windows message box
+  def Mbox(title, text, style):
+    return ctypes.windll.user32.MessageBoxW(0, text, title, style)
+
+  # Clears the console
+  def clearConsole():
+    clr = lambda: os.system('cls' if os.name in ('nt', 'dos') else 'clear')
+    return clr
```
Comment (itzCozi) [inline source/converter.py:52]: Add functions class to contain all functions

File: source/converter.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+class colors:
+  '''
+  This is a class containing variables of similar values
+  To use this class call the class name then the variable 
+  '''
+  blue = os.system("color b")
+  red = os.system("color C")
+  green = os.system("color a")
+  pink = os.system("color D")
+
```
Comment (itzCozi) [inline source/converter.py:41]: Add colors class for all colors variables

File: source/converter.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
```
Comment (itzCozi) [inline source/converter.py:42]: This is called a doc-string and its for displaying informatation about the function or class in plain text.

File: source/converter.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    option_text = "Choose your Converter > "
+
+    functions.clearConsole()
+    print(banner)
+    colors.pink
+
+    # Use '\n' to create a new line
+    print("[1] YouTube to MP4 \n")
+    print("[2] Image Downloader \n")
+    print("[3] YouTube to MP3 (New!) \n\n")
```
Comment (itzCozi) [inline source/converter.py:76]: Set a variable for commonly used texts like this one here

File: source/converter.py [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+  	  * Used while True instead of while option != 0 to simplify the loop.
+		  * Used input() instead of int(input()) to handle non-integer input gracefully.
+		  * Used option.lower() to convert the input option to lowercase and simplify the if statements.
+		  * Removed the exit() calls in the try-except blocks and used continue instead to keep the loop running.
+	    * Added better exit() mechanic with sys.exit() which passing a 1 means an error occured.
+		  * Added a prompt for the destination directory in option 3.
+		  * Moved the menu() call outside the function to prevent an infinite recursion loop.
+    '''
+
+    option_text = "Choose your Converter > "
```
Comment (itzCozi) [inline source/converter.py:67]: Ensure to set commonly used strings as a variable.

File: source/converter.py [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+             'Thanks for Using This Tool :)', 1)
+        sys.exit(0)
+
+      else:
+        print("Invalid option!")
+        continue
+
+
+# This is the first block of code ran
+try:
```
Comment (itzCozi) [inline source/converter.py:150]: This is called driver code and it is the starter for the entire program

**Comment by Crafttino21** [comment]
Thanks. I removed the clear/cls block from you because I got a conflict in the code what crashes the program. Thanks for a fix 👌🏻

**Comment by Crafttino21** [comment]
Thanks for you help. I am still working on it

**Comment by itzCozi** [comment]
Big thanks man ill help whenever I can and take a look at clearConsole() when I get home!

**Pull request merged**